### PR TITLE
FSDP: Mistral(mathstral) sbatch file - MISTRAL MODEL SUPPORT

### PR DESCRIPTION
**Pull request by nithiyn**
Adding support for Mistral models when using FSDP on Hyperpod. Previously only mixtral was supported. This sbatch file is part of a hyperpod blog on training mathstral. This PR is part of other PRs that Armando and I will incorporate as part of the mathstral sbatch job already tested on Hyperpod.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: 3.test_cases/10.FSDP/3.distributed-training-mistral-mathstral.sbatch [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+############################
+# Mixtral Training Params ##
+############################
+
+declare -a TRAINING_ARGS=(
+    --train_batch_size=1 \
+    --val_batch_size=1 \
+    --max_steps=5000 \
+    --seed=42 \
+    # torch.get_default_dtype() returns fp32
```
Comment (perifaws) [inline 3.test_cases/10.FSDP/3.distributed-training-mistral-mathstral.sbatch:62]: remove

Comment (nithiyn) [reply 3.test_cases/10.FSDP/3.distributed-training-mistral-mathstral.sbatch:62]: hey Pierre, would you like us to just remove just the comment with fp32 or the other lines you've highlighted as well?

Comment (perifaws) [reply 3.test_cases/10.FSDP/3.distributed-training-mistral-mathstral.sbatch:62]: comment only :) The thing is that either you keep it and document why it's there and commented or remove. typically these are left-overs of coding or left for a good reason but in which case it needs a bit more info

Comment (nithiyn) [reply 3.test_cases/10.FSDP/3.distributed-training-mistral-mathstral.sbatch:62]: Understood! Thanks for the quick response

Comment (nithiyn) [reply 3.test_cases/10.FSDP/3.distributed-training-mistral-mathstral.sbatch]: Have removed the comment

**Comment by perifaws** [comment]
Hey @nithiyn , you'll want to amend the [readme](https://github.com/aws-samples/awsome-distributed-training/tree/main/3.test_cases/10.FSDP#0-prerequisites) file. For example, you could add a subsection in the **3.Launch Training** to show how to run this new case (example of correct output for example).

**Comment by nithiyn** [comment]
> Hey @nithiyn , you'll want to amend the [readme](https://github.com/aws-samples/awsome-distributed-training/tree/main/3.test_cases/10.FSDP#0-prerequisites) file. For example, you could add a subsection in the **3.Launch Training** to show how to run this new case (example of correct output for example).

sorry for the late response, somehow missed this but yes, can do

**Comment by nithiyn** [comment]
> Hey @nithiyn , you'll want to amend the [readme](https://github.com/aws-samples/awsome-distributed-training/tree/main/3.test_cases/10.FSDP#0-prerequisites) file. For example, you could add a subsection in the **3.Launch Training** to show how to run this new case (example of correct output for example).

Hey @perifaws, have updated the readme to include mathstral and the output when launching training, please let me know if there are any other changes required. Thanks!

**Review by perifaws — APPROVED** [review]

**Pull request merged**